Fix register_instance from CLI

pirlgon opened the pull request:
An UserError is raised when environment does not exist. Environment load has been set to joined for Instance.

pirlgon commented:
fix #1530

bersace commented on ui/temboardui/model/orm.py:
```diff
@@ -348,7 +348,7 @@ class Instance(Model):
     )
 
-    environment = relationship("Environment", back_populates="instances", lazy="raise")
+    environment = relationship("Environment", back_populates="instances", lazy="joined")
```
Hmm, je souhaite des join explicites pour se débarrasser un jour de l'ORM. Pourquoi faut-il un join implicite ?

pirlgon replied:
On a une erreur quand on fait le register car on essai d'afficher les infos de l'environnement de l'instance mais il n'est pas chargé.

pirlgon replied:
```
08:33:51 temboardui[1017037] ERROR:  app: Unhandled error:
	Traceback (most recent call last):
	  File "/usr/lib/python3.9/site-packages/temboardui/toolkit/app.py", line 312, in entrypoint
	    retcode = self.main(argv, environ)
	  File "/usr/lib/python3.9/site-packages/temboardui/cli/app.py", line 116, in main
	    return command.main(args)
	  File "/usr/lib/python3.9/site-packages/temboardui/cli/register_instance.py", line 213, in main
	    self.output_instance(instance)
	  File "/usr/lib/python3.9/site-packages/temboardui/cli/register_instance.py", line 218, in output_instance
	    json.dump(instance.asdict(), sys.stdout, indent="  ", cls=JSONEncoder)
	  File "/usr/lib/python3.9/site-packages/temboardui/model/orm.py", line 489, in asdict
	    environment=self.environment.name,
	  File "/usr/lib64/python3.9/site-packages/sqlalchemy/orm/attributes.py", line 482, in __get__
	    return self.impl.get(state, dict_)
	  File "/usr/lib64/python3.9/site-packages/sqlalchemy/orm/attributes.py", line 942, in get
	    value = self._fire_loader_callables(state, key, passive)
	  File "/usr/lib64/python3.9/site-packages/sqlalchemy/orm/attributes.py", line 978, in _fire_loader_callables
	    return self.callable_(state, passive)
	  File "/usr/lib64/python3.9/site-packages/sqlalchemy/orm/strategies.py", line 856, in _load_for_state
	    self._invoke_raise_load(state, passive, "raise")
	  File "/usr/lib64/python3.9/site-packages/sqlalchemy/orm/strategies.py", line 816, in _invoke_raise_load
	    raise sa_exc.InvalidRequestError(
	sqlalchemy.exc.InvalidRequestError: 'Instance.environment' is not available due to lazy='raise'

```

pirlgon committed:
Test register-instance CLI

pirlgon committed:
Fix register_instance from CLI

An UserError is raised when environment does not exist.
We also fix missing space in log message.

bersace commented on tests/test_20_inventory.py:
```diff
@@ -39,6 +39,16 @@ def test_web_register(registered_agent, agent_conf, browser, pg_version, ui_url)
     assert pg_version in browser.select(fmt(col=postgres)).text
```
Titre de commit: "Test register-instance CLI"

bersace commented on CHANGELOG.md:
```diff
@@ -5,6 +5,11 @@ CI release job extracts changes from this file and attaches them as GitHub relea
 
+## UNREALESED
+
+- Fix register instance from temboard cli.
```
`Fix register-instance CLI`

Deuxième item: `Raise proper error on unknown environment`.

bersace commented on CHANGELOG.md:
```diff
@@ -5,6 +5,11 @@ CI release job extracts changes from this file and attaches them as GitHub relea
 Ensure you use consistent title format.
 -->
 
+## UNREALESED
```
*UNRELEASED*
